### PR TITLE
Don’t use glob expansion in argument to `cd` in updater.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

As in the summary. We should never be using glob expansion in arguments for commands that expect a single argument.

In this case, we can use a combination of `find` and `head` to extract exactly one matching directory based on some additional info we already have, and then pass just that path to `cd`.

##### Component Name

area/packaging

##### Test Plan

Verified using CI.

##### Additional Information

Fixes: #10931

### CODE REVIEW

**Review by vkalintiris — APPROVED** [review]
Not related to this PR, but it would be awesome if we could start using `set -u` in our functions/scripts.

**Review by knatsakis — APPROVED** [review]

**Review by vkalintiris — APPROVED** [review]

**Comment by Steve8291** [comment]
The `find` command looks like it will work to fix #10937
Closing #10937 in favor of this fix.
I also agree about using `set -u` just need to check all the scripts to make sure that they don't expect to continue if a variable isn't set. I've been caught by that before.

**Comment by Steve8291** [comment]
One last thought on this - Will users have to manually run the `kickstart.sh` again if they have the older non-working code from this `netdata-updater.sh` installed? I know my installation had stopped updating altogether once it downloaded the non-functioning `cd` command.

**Comment by Ferroin** [comment]
> One last thought on this - Will users have to manually run the `kickstart.sh` again if they have the older non-working code from this `netdata-updater.sh` installed? I know my installation had stopped updating altogether once it downloaded the non-functioning `cd` command.

The `netdata-updater.sh` script will attempt to run the newest code from the repo if it’s newer than the local copy, and that check and handoff happens before the problematic piece of code gets run. As a result, as long as this is not masking some other issue (and for all I know it might be), things should start working as soon as this gets merged.

File: packaging/installer/netdata-updater.sh [hunk 1]
```diff
@@ -295,7 +295,7 @@ update() {
     NEW_CHECKSUM="$(safe_sha256sum netdata-latest.tar.gz 2> /dev/null | cut -d' ' -f1)"
     tar -xf netdata-latest.tar.gz >&3 2>&3
     rm netdata-latest.tar.gz >&3 2>&3
-    cd netdata-* || exit 1
+    cd "$(find . -maxdepth 1 -name "netdata-${path_version}*" | head -n 1)" || exit 1
```
Comment (nabijaczleweli) [inline packaging/installer/netdata-updater.sh:298]: `-print -quit`

**Pull request merged**